drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -49,34 +50,41 @@ config ETH_STM32_HAL_PHY_ADDRESS
 	  The phy address to use.
 
 config ETH_STM32_HAL_RANDOM_MAC
-	bool "Random MAC address"
+	bool "Random MAC address (DEPRECATED)"
 	depends on ENTROPY_GENERATOR
-	default y
+	select DEPRECATED
 	help
 	  Generate a random MAC address dynamically.
+	  Deprecated in favor of device tree property "zephyr,random-mac-address"
 
 if !ETH_STM32_HAL_RANDOM_MAC
 
 config ETH_STM32_HAL_MAC3
-	hex "MAC Address Byte 3"
+	hex "MAC Address Byte 3 (DEPRECATED)"
 	default 0
 	range 0 0xff
+	select DEPRECATED
 	help
 	  This is the byte 3 of the MAC address.
+	  Deprecated in favor of device tree property "local-mac-address"
```

Comment:
Actually yes, this would always generate a deprecated warning, something doesn't seem right here, maybe use a choice

Reply:
@nordicjm 
Also note that `select DEPRECATED` can't be used with these hex symbols:
```
warning: the hex symbol ETH_STM32_HAL_MAC3 (defined at drivers/ethernet/Kconfig.stm32_hal:62) has selects or implies
```
In the end, I don't see a simple change that would not add a new symbol, nor break user configuration.

What about the initial, efficient and simple:
```
#if CONFIG_ETH_STM32_HAL_MAC3 || \
	CONFIG_ETH_STM32_HAL_MAC4 || \
	CONFIG_ETH_STM32_HAL_MAC5
#warning CONFIG_ETH_STM32_HAL_MAC* is deprecated, use device tree "local-mac-address"
```
?

Reply:
@nordicjm I reverted to the warning solution. Please tell me if you have better idea.